### PR TITLE
Api 19397 regex for v2

### PR DESCRIPTION
**Pull request by Code4FunXD**

### CODE REVIEW

**Comment by Code4FunXD** [comment]
Right now I have the regex within the path values of some of the older controllers and was wondering if we want it this way. I have another solution for the regex addition which would be to have a split of the endpoints of v1 and v2 like this:

 @GetMapping({"/v1/disability_rating/{icn}"})
  public DisabilityRatingResponse findRatingDataV1Response(@NonNull @PathVariable("icn") String icn) {
    ...
  }

  @GetMapping({"/v2/disability_rating/{icn}"})
  public DisabilityRatingResponse findRatingDataV2Response(@NonNull @PathVariable("icn") @Pattern(regexp = "^\\d{10}V\\d{6}$") @Size(min = 17, max = 17)
   ... 
 }
 
 I think this is probably a better option since we will get constraint violations as an exception. What you guys think?

**Comment by dlennon-va** [comment]
> Right now I have the regex within the path values of some of the older controllers and was wondering if we want it this way. I have another solution for the regex addition which would be to have a split of the endpoints of v1 and v2 like this:
> 
> ```
> @GetMapping({"/v1/disability_rating/{icn}"})
>   public DisabilityRatingResponse findRatingDataV1Response(@NonNull @PathVariable("icn") String icn) {
>     ...
>   }
> 
>   @GetMapping({"/v2/disability_rating/{icn}"})
>   public DisabilityRatingResponse findRatingDataV2Response(@NonNull @PathVariable("icn") @Pattern(regexp = "^\\d{10}V\\d{6}$") @Size(min = 17, max = 17)
>    ... 
>  }
> ```
> 
> I think this is probably a better option since we will get constraint violations as an exception. What you guys think?

I'd say just do the `@Pattern` and not separate the endpoints. /v1 and /v2 can both have the pattern and thats fine.

**Comment by Code4FunXD** [comment]
> > Right now I have the regex within the path values of some of the older controllers and was wondering if we want it this way. I have another solution for the regex addition which would be to have a split of the endpoints of v1 and v2 like this:
> > ```
> > @GetMapping({"/v1/disability_rating/{icn}"})
> >   public DisabilityRatingResponse findRatingDataV1Response(@NonNull @PathVariable("icn") String icn) {
> >     ...
> >   }
> > 
> >   @GetMapping({"/v2/disability_rating/{icn}"})
> >   public DisabilityRatingResponse findRatingDataV2Response(@NonNull @PathVariable("icn") @Pattern(regexp = "^\\d{10}V\\d{6}$") @Size(min = 17, max = 17)
> >    ... 
> >  }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think this is probably a better option since we will get constraint violations as an exception. What you guys think?
> 
> I'd say just do the `@Pattern` and not separate the endpoints. /v1 and /v2 can both have the pattern and thats fine.

Story says not to do for v1 [API-19397](https://vajira.max.gov/secure/RapidBoard.jspa?rapidView=6240&view=detail&selectedIssue=API-19397). Do you think it'd be fine tho?

**Comment by amizan8653-va** [comment]
tl;dr: I wanna do David's approach of just sticking the `@Pattern` in there to apply constraint to both V1 & V2. 
-------

I like David's suggestion, but if we *have* to avoid adding the regex constraint to V1 endpoints, then it means we can't do that. In that case, I like having the controller methods split out. 

But... I think even though the ticket states that we shouldn't add the regex param constraint to V1 endpoints, I think it should be fine to the existing consumers if we add the parameter constraint. Without the constraint: If a consumer was passing in a bad request, they'd get a 500 internal server error as we try to move forward processing bad data. With the parameter constraint added to V1, the v1 endpoints would still get an error code. It would just be a more accurate 400 bad request instead of a 500 internal server error.

**Review by amizan8653-va — APPROVED** [review]
